### PR TITLE
Fix Swiftlint on Xcode 12.5

### PR DESCRIPTION
**Pull request by whiskey**
## Description
Swiftlint did not trigger some warnings when using Xcode 12.5/ Swift 5.4.
This PR fixes this by using a manually compiled branch of Swiftlint.

* It also tweaks the output format of the linter to simplify review in the CI
* The experimental (static code) "analyze" feature has been added to fastlane but is currently not automatically used. 

## Link to Jira
n/a

## Screenshots
Opening the artifacts in CI brings up the HTML report:
Screen Shot 2021-06-17 at 08 42 54
Screen Shot 2021-06-16 at 17 10 11

Analyze:
Screen Shot 2021-06-17 at 07 06 26
Screen Shot 2021-06-17 at 07 06 38

### CODE REVIEW

**Comment by whiskey** [comment]
For the `analyze` thing I would like to get these reports back into Xcode but struggle to do this right now. I'm open to any suggestions: https://stackoverflow.com/questions/68014062/display-swiftlint-analyze-results-in-xcode.

File: src/xcode/fastlane/Fastfile [hunk 1]
```diff
@@ -18,6 +18,8 @@
 ENV["LC_ALL"] = "de_DE.UTF-8"
 ENV["LANG"] = "de_DE.UTF-8"
 
+ENV["FASTLANE_SKIP_UPDATE_CHECK"] = "True"
```
Comment (pascal-brause) [inline src/xcode/fastlane/Fastfile:21]: Do we get noticed ever when we have to update fastlane?

File: src/xcode/fastlane/Fastfile [hunk 2]
```diff
@@ -63,13 +65,40 @@ platform :ios do
...
+    )
+    swiftlint(
+      executable: "../../vendor/swiftlint",
+      mode: :analyze,
+      compiler_log_path: "DerivedData/logs/ENA-ENA.log",
+      output_file: "swiftlint.analyze",
+      config_file: "ENA/.swiftlint.yml",
+      strict: false,
       quiet: false,
-      raise_if_swiftlint_error: true,
```
Comment (pascal-brause) [inline src/xcode/fastlane/Fastfile:71]: does this mean a linter error does not raise a failure?

**Review by pascal-brause — COMMENTED** [review]
I understand why you did this but when we use a manually compiled version of SwiftLint, do we have to maintain this manually when ever a new SwiftLint version is released?

File: src/xcode/fastlane/Fastfile [hunk 3]
```diff
@@ -63,13 +65,40 @@ platform :ios do
...
-      reporter: "sonarqube"
```
Comment (pascal-brause) [inline src/xcode/fastlane/Fastfile:72]: Does this mean we replace sonarquebe by SwiftLint analyze?

File: src/xcode/fastlane/Fastfile (hunk 1 above)
Comment (whiskey) [reply src/xcode/fastlane/Fastfile:21]: Nobody cares on the CI and I run fastlane manually more than once per day. I'll get notice about updates and tbh, we don't have to get every dot update for fastlane.

File: src/xcode/fastlane/Fastfile (hunk 2 above)
Comment (whiskey) [reply src/xcode/fastlane/Fastfile:71]: I think I missed this when I was working on a better lint output…

File: src/xcode/fastlane/Fastfile (hunk 3 above)
Comment (whiskey) [reply src/xcode/fastlane/Fastfile:72]: Nope. It's still experimental, takes ~1 hour to compile a report and does not simply includes into Xcode. It's fun to scroll through the local list of analyzer issues though.

Comment (whiskey) [reply src/xcode/fastlane/Fastfile:72]: And for Cube, we never used that report.

File: src/xcode/fastlane/Fastfile (hunk 1 above)
Comment (whiskey) [reply src/xcode/fastlane/Fastfile:21]: About the static SwiftLint: It's a hack until we have a new release that covers this issue in Xcode 12.5.

File: src/xcode/fastlane/Fastfile (hunk 2 above)
Comment (whiskey) [reply src/xcode/fastlane/Fastfile:71]: @pascal-brause I tested the current setup and it fails fine on linter errors: https://app.circleci.com/pipelines/github/corona-warn-app/cwa-app-ios/17289/workflows/e80b657e-aed2-4255-b401-22e839be3c55/jobs/57223

**Review by marcussc — APPROVED** [review]

**Review by ArturFriesen — CHANGES_REQUESTED** [review]
"using a manually compiled branch of Swiftlint"
This worries me.
Is this temporary?
If not, I would like to talk about that in the DevCamp.

**Review by ArturFriesen — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**